Do not depend on cobertura plugin

Even when the "Uberalls" checkbox is not set, the build will fail if the cobertura plugin is not installed in Jenkins.

```bash
[phabricator:plugin-provider] 'cobertura' plugin not installed.
ERROR: Step ‘Post to Phabricator’ aborted due to exception: 
java.lang.NoClassDefFoundError: hudson/plugins/cobertura/CoberturaBuildAction
	at com.uber.jenkins.phabricator.coverage.CoberturaCoverageProvider.getCoverageResult(CoberturaCoverageProvider.java:89)
	at com.uber.jenkins.phabricator.coverage.CoberturaCoverageProvider.hasCoverage(CoberturaCoverageProvider.java:46)
	at com.uber.jenkins.phabricator.PhabricatorNotifier.getCoverageProvider(PhabricatorNotifier.java:216)
	at com.uber.jenkins.phabricator.PhabricatorNotifier.perform(PhabricatorNotifier.java:89)
	at hudson.tasks.BuildStepMonitor$1.perform(BuildStepMonitor.java:20)
	at hudson.model.AbstractBuild$AbstractBuildExecution.perform(AbstractBuild.java:782)
	at hudson.model.AbstractBuild$AbstractBuildExecution.performAllBuildSteps(AbstractBuild.java:723)
	at hudson.model.Build$BuildExecution.post2(Build.java:185)
	at hudson.model.AbstractBuild$AbstractBuildExecution.post(AbstractBuild.java:668)
	at hudson.model.Run.execute(Run.java:1763)
	at hudson.model.FreeStyleBuild.run(FreeStyleBuild.java:43)
	at hudson.model.ResourceController.execute(ResourceController.java:98)
	at hudson.model.Executor.run(Executor.java:410)
Caused by: java.lang.ClassNotFoundException: hudson.plugins.cobertura.CoberturaBuildAction
	at jenkins.util.AntClassLoader.findClassInComponents(AntClassLoader.java:1376)
	at jenkins.util.AntClassLoader.findClass(AntClassLoader.java:1326)
	at jenkins.util.AntClassLoader.loadClass(AntClassLoader.java:1079)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:358)
```

--- a/changelog.md
+++ b/changelog.md
@@ -4,6 +4,7 @@
 
 * Add more logging on differential fetch failure
 * Allow user to apply patch with force flag (Chaitanya Pramod)
+* Fix crash when missing cobertura plugin
 
 ### 1.8.3 (2015/12/09)
 

--- a/src/main/java/com/uber/jenkins/phabricator/provider/InstanceProvider.java
+++ b/src/main/java/com/uber/jenkins/phabricator/provider/InstanceProvider.java
@@ -55,6 +55,7 @@ public class InstanceProvider<T> {
     public T getInstance() {
         if (!provider.isAvailable()) {
             logger.info(LOGGER_TAG, String.format("'%s' plugin not installed.", pluginName));
+            return null;
         }
         T instance = provider.getInstance(className);
         if (instance == null) {

--- a/src/test/java/com/uber/jenkins/phabricator/provider/InstanceProviderTest.java
+++ b/src/test/java/com/uber/jenkins/phabricator/provider/InstanceProviderTest.java
@@ -39,6 +39,12 @@ public class InstanceProviderTest {
     }
 
     @Test
+    public void testUnavailablePluginValidClass() {
+        InstanceProvider<UnitTestProvider> provider = makeProvider("weird-plugin-name", "com.uber.jenkins.phabricator.unit.JUnitTestProvider");
+        assertNull(provider.getInstance());
+    }
+
+    @Test
     public void testBadClassName() {
         InstanceProvider<UnitTestProvider> provider = makeProvider("junit", "com.nonexistent.class");
         assertNull(provider.getInstance());

